### ISSUE TITLE
Refinements to `collapse_to_higher_taxon`

### ISSUE THREAD

**Issue by ehwenk** [traitecoevo/APCalign#127]
We should discuss and possibly refine the functionality of the `collapse_to_higher_taxon` option in `create_taxonomoic_update_lookup`.

First, the output name of the collapsed name is `apc_names`. 

Do we want this information to overwrite `suggested_names` or is it a separate column? I assume overwriting `suggested_names`.

Second, in the example below, where there are infraspecific and species-level names within a single genus, it collapses the infraspecific names to the species, but then doesn't collapse other names with Cycas.

Is it possibly instead that we actually want it to just collapse names that are ambiguous due to splits? 

Genus- and family-level alignments are already available as standalone columns if that is the output people want. In AusTraits we also have a column where infraspecific taxa are collapsed to species, but I don't think we really want to add this to APCalign.
```
taxa <- c("Cycas calcicola", "Cycas canalis subsp. carinata", "Cycas canalis subsp. canalis", "Cycas canalis", "Cycas candida", "Cycas conferta", "Centotheca malabarica", "Catapodium loliaceum")
```

**Comment by wcornwell** [comment]
first part of the problem is that the column names need to get updated here: https://github.com/traitecoevo/APCalign/blob/c1001e967922279aac1493be1ae11dc6261884d4/R/create_taxonomic_update_lookup.R#L138

second part might be a conflict with earlier changes.  i'll assess after the merge.

**Comment by wcornwell** [comment]
Current behavior.  It's a bit weird that unknown names are getting collapsed to plants.  Is it better to leave them as NA? 
<img width="877" alt="Screenshot 2023-08-22 at 8 57 34 am" src="https://github.com/traitecoevo/APCalign/assets/5534161/8de0d595-fc58-44a5-b6a0-3410cc7c9426">

**Comment by wcornwell** [comment]
this seems like a bit weird: 
<img width="876" alt="Screenshot 2023-08-22 at 9 00 29 am" src="https://github.com/traitecoevo/APCalign/assets/5534161/9f32f2e0-f1a0-477b-a877-c3f9a8e5c091">

**Comment by fontikar** [comment]
What is the expected behaviour for`"Centotheca malabarica", "Catapodium loliaceum"` when one_to_many = collapse to higher taxon?

**Comment by wcornwell** [comment]
I'm not sure, but i think `NA` would probably be better?

**Comment by ehwenk** [comment]
The current coding changes @dfalster and I have made mean that the same number of rows is now returned, but, for taxa that *need* collapsing, a higher-level name is returned. This is actually more logical to me, because it is showing a "collapsed" output for each input name. @wcornwell what do you think?

**Comment by ehwenk** [comment]
@wcornwell @dfalster 

We need to decide what outcome we actually want for this option. To me, it should only collapse taxa with actual ambiguity, such that the number of rows output by `create_taxonomic_update_lookup` should be the same as for the `most_likely_species` option, but with different suggested names for ambiguous taxa, an added column for `number of taxa collapsed`, and a column listing all names collapsed. 

For instance:
- `Acacia aneura` becomes `Acacia sp.` [number collapsed = 3], but `Acacia minyura` remains `Acacia minyura` [number collapsed = 1]. 
- Anything with `taxon_rank` = genus, remains `taxon_rank = genus` and its suggested name becomes/remains `genus sp.` [number collapsed = 1]
- Anything with `taxon_rank` = family, remains `taxon_rank = family` and its suggested name becomes/remains `family sp.` [number collapsed = 1]
- Just because `Acacia aneura` is in the original list of taxon names, does not affect other `Acacia` entries in the original list - they retain their species status/name if there isn't any ambiguity.

If you agree, then this becomes a variant on the code that creates splits and selects most likely species, rather than a standalone function.

**Comment by wcornwell** [comment]
I agree. I think we should say explicitly that we only do taxonomic updating at the species level or below. And higher level updating is outside of scope.

**Comment by ehwenk** [comment]
I was thinking some more about what names should be reformatted. And now I'm leaning towards leaving those names that are already declared to be 'genus' or 'family' level by align_taxa as they are. The taxon_rank is already genus (or family) and I don't think people want to lose the notes in [] associated with those names just because they want splits to be collapsed. This also allows me to move more code into the APC species sub function - as @dfalster suggested. But that leads to a second thought. What is genus-level alignments due to collapsing as formatted the same as we do for sp aff, intergrades, hybrids, and format them as 'genus sp. [names of collapsed taxa]' . That saves us from having a special column to document collapsed names and means it is in the same format as the names that were effectively collapsed by the alignment process?

**Comment by ehwenk** [comment]
Closed with commit ae058ec8
